src/sentry/static/sentry/app/utils/profiler.jsx

```diff
@@ -1,56 +1,83 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+import {debounce} from 'lodash';
+import React, {unstable_Profiler as ReactProfiler} from 'react';
 import * as Sentry from '@sentry/browser';
 
 import {startRender, finishRender} from 'app/utils/apm';
 import getDisplayName from 'app/utils/getDisplayName';
 
-export default function profiler() {
-  return WrappedComponent => {
-    const displayName = getDisplayName(WrappedComponent);
+class Profiler {
+  spansById = new Map([]);
 
-    return class extends React.Component {
-      static displayName = displayName;
+  dispose() {
+    this.spansById = new Map([]);
+  }
 
-      static propTypes = {
-        api: PropTypes.object,
-      };
+  commit = debounce((id, span, timestamp) => {
+    span.timestamp = timestamp;
+    span.finishedSpans.push(span);
+    Sentry.finishSpan(span);
+    finishRender(id);
+  }, 500);
 
-      componentWillUnmount() {
-        this.finishProfile();
-      }
+  handleRender = (
+    id,
+    phase,
+    actualDuration,
+    _baseDuration,
+    startTime,
+    _commitTime,
+    _interactions
+  ) => {
+    let span;
+    const start = window.performance && window.performance.timing.navigationStart;
```

Comment:
@billyvg According to the docs, the `PerformanceTiming` interface is being deprecated: https://developer.mozilla.org/en-US/docs/Web/API/PerformanceTiming in favor of `PerformanceNavigationTiming`.